SpatialGDK/Source/SpatialGDK/Public/Utils/SchemaOption.h

```diff
@@ -104,7 +104,7 @@ class TSchemaOption
 };
 
 template <typename T>
-inline uint32 GetTypeHash(const improbable::TSchemaOption<T>& Option)
+inline uint32 GetTypeHash(const SpatialGDK::TSchemaOption<T>& Option)
```

Comment:
I think this actually may be broken (and was already broken before the rename) since `GetTypeHash` needs to be in the global namespace and here it looks like it's in `SpatialGDK`. I guess we never used `TSchemaOption` as a key in a `TMap` etc.

Reply:
Looks like you've addressed this in your PR.

Reply:
Looked into this yesterday and discussed slightly offline - I think GetType has explicitly can't be in the global namespace. Can go through this with someone and work out exactly where it should be for a dependent template lookup.

Reply:
So should I wait to merge this? What's the issue with GetTypeHash being in the global namespace?

Reply:
This should be good to merge for now since we never use `TSchemaOption` as a key in `TMap`s. We might need to revisit this later if we ever intend to start using it that way.

Reply:
GetTypeHash is used in a load of template magic in determining if the type used as a key in TMap or TSet has a hash function.
It's been implemented by checking if the function returns something other than a placeholder type rather than a check on if any such function exists, so you need the function we define to be found ahead of the overload in Unreal.

So it's basically making sure that ADL searches in the right place first. In this case I think the global namespace will always lose to their overload. I think we'd get the same issue if we tried to define a hash function in some other namespace that had to find the right overload for this type.

Reply:
TL;DR I think this is fine as it is.